script.js

```diff
@@ -0,0 +1,119 @@
+    link.appendChild(image); //append image as the last child of an <a> tag
+    gallery.appendChild(link); //append <a> tag as the last child of the gallery
+  },
+  createImage: function(imageName, imageSrc) {
+    var loadedImage = document.createElement("img"); //create new image object
+    loadedImage.setAttribute("height", imageProcessor.parameters.thumbHeight); //add attributes to the image: size, title, source
+    loadedImage.setAttribute("width", imageProcessor.parameters.thumbWidth);
+    loadedImage.setAttribute("title", imageName);
+    loadedImage.setAttribute("src", imageSrc);
+    return loadedImage;
+  },
+  createLink: function(imageSrc) {
+    var loadedLink = document.createElement("a"); //create <a> tag for hyperlink
+    loadedLink.setAttribute("target", "_blank"); //image will open in new window
+    loadedLink.setAttribute("href", imageSrc); //URL of the page
+    return loadedLink;
+  }
+}
+
+var eventHandlerModule = {
+  attachEvent: function(target, event, callback) {
+    target.addEventListener(event, callback, false);
+  },
+  detachEvent: function(target, event, callback) {
+    target.removeEventListener(event, callback, false);
+  }
+}
+
+var fileModule = {
+  handleFiles: function(querySel) {
+    var files = document.querySelector(querySel).files; //variable, which shows how many files is loading
+    if (files) { 
+      [].forEach.call(files, fileModule.reading); //for each file, which you load is made loadImage function (multiple files)
+    }
+  },
+  reading: function(image) {
+    var reader = new FileReader(); //create new FileReader(), which lets read the content of files
+    var regex = ".(" + imageProcessor.parameters.allowedFiles.join("|") + ")$"; //regular expression made from configuration parameters
+    var regExp = new RegExp(regex, "i"); //make new RegExp with ignoring letter case
+    if (regExp.test(image.name)) { //condition for images format
+      eventHandlerModule.attachEvent(reader, "load", function() {
+        galleryModule.makeGallery(image.name, this.result);
+      });
+      reader.readAsDataURL(image);
+    }
+  }
+}
+
+var imageProcessor = {
+  parameters: {
+    thumbWidth: 150,
+    thumbHeight: 150,
+    dropArea: '#dropArea',
+    input: '#fieldInput',
+    allowedFiles: ['jpg', 'png']
+  },
+
+  init: function(arg) {
+    if (arg !== undefined) { //update when new parameters are gave to function
+      function updateConfig() {
+        for (var prop in arg) {
+          for (var propert in imageProcessor.parameters) {
+            if (prop === propert) {
+              imageProcessor.parameters[propert] = arg[prop];
+            }
+          }
+        }
+      }
+      updateConfig();
+      //imageProcessor.parameters = arg; //we can use this line rather than updateConfig() if configuration can be updated only with all properties
+    }
+    function createHTML() {

```

Comment:
ok, rozumiem, wcześniej myślałam, że właśnie ma powstać to wszystko w js zgodnie z konfigiem, a nie odwrotnie